logos/benchmarks/benchmark_logos.py

```
@@ -3109,60 +3057,65 @@ def _ollama_compose_content(models_dir: str, local_models_dir: str) -> str:
 
 
-def _deploy_ollama_compose_via_ssh(
+def _start_sllm_head(compose_dir: str, models_dir: str, use_sudo: bool) -> None:
+    """Start the SLLM head node locally via docker compose."""
+    sudo = "sudo " if use_sudo else ""
+    os.makedirs(compose_dir, exist_ok=True)
+    compose_file = os.path.join(compose_dir, "docker-compose.yml")
+    with open(compose_file, "w") as fh:
+        fh.write(_sllm_head_compose_content(models_dir))
+    print(f"  [sllm] Starting head node (API port {_SLLM_API_PORT}) ...")
+    result = subprocess.run(
```

Ruff (0.15.18)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)